Acceptable-Formulae: update static library guidance. (#16656)
Homebrew/brew: docs/Acceptable-Formulae.md

We're a package manager so we want to do things like resolve dependencies and set up applications for our users. If things require too much manual intervention then they aren't useful in a package manager.

### Shared vs. static libraries

In general, if formulae have to ship either shared or static libraries: they should ship shared ones.

> If they can ship both, they should.

Reviewer: That's introducing a bunch of extra code and maintenance. Let's just allow it and not mandate it

Author: @SMillerDev Can you suggest a rewording?
```suggestion
If they can ship both, they can.
```

?

Reviewer: I'd say keeping the current situation is fine
```suggestion
If there is demand for a static library they can ship both.
```

Author: @SMillerDev I think whenever we talk about "demand" we tend to imply "by more than one person" which I don't think is/should be the case here. I get the desire to say that it's not a requirement but I also would rather we softened the language to make it clear we welcome these PRs.

Reviewer: How about `s/demand/interest`

Author: Yup, that works.

> If there is interest in a static library they can ship both.

Reviewer 2: Nit: a formula can provide many libraries. Also for consistency.

```suggestion
If there is interest in static libraries they can ship both.
```

> Shipping only static libraries should be avoided when possible, particularly when the formulae is depended on by other formulae since these dependents cannot be updated without a rebuild.

Reviewer 2: ```suggestion
Shipping only static libraries should be avoided when possible, particularly when the formula is depended on by other formulae since these dependents cannot be updated without a rebuild.
```
